src/extensions/rich-text/paste-markdown.ts

```diff
@@ -66,27 +69,42 @@ const PasteMarkdown = Extension.create({
+                            vsCodeClipboardMetadata.mode !== 'markdown'
+
+                        const isInsideCodeBlockNode =
+                            editor.state.selection.$from.parent.type.name === 'codeBlock'
+
                         // Do not handle the paste event if:
-                        //  * The clipboard doesn't contain plain-text
-                        //  * The clipboard HTML content doesn't come with VS Code metadata
-                        //  * The clipboard VS Code metadata does not represent plain-Markdown
-                        //  * The user is pasting inside a code block element
+                        //  * The clipboard does NOT contain plain-text
+                        //  * The clipboard contains HTML but from an unknown source (like Google
+                        //    Drive, Dropbox Paper, etc.)
+                        //  * The clipboard contains HTML from VS Code that it's NOT plain-text or
+                        //    Markdown (like Python, TypeScript, JSON, etc.)
+                        //  * The user is pasting content inside a code block node
+                        // For all the above conditions we want the default handling behaviour from
+                        // ProseMirror to kick-in, otherwise we'll handle it ourselves below
                         if (
-                            !clipboardText ||
-                            (clipboardContainsHTML && !clipboardContainsVSCodeMetadata) ||
-                            (clipboardContainsVSCodeMetadata &&
-                                vsCodeClipboardMetadata.mode !== 'markdown') ||
-                            editor.state.selection.$from.parent.type.name === 'codeBlock'
+                            !clipboardContainsText ||
+                            clipboardContainsHTMLFromUnknownSource ||
+                            clipboardContainsHTMLFromVSCodeOtherThanTextOrMarkdown ||
+                            isInsideCodeBlockNode
                         ) {
                             return false
                         }
 
                         // Send the clipboard text through the HTML serializer to convert potential
                         // Markdown into HTML, and then insert it into the editor
                         editor.commands.insertMarkdownContent(
-                            // The slice content is used instead of `clipboardText` as the pasted
-                            // content could have already been transformed by other plugins
-                            slice.content.textBetween(0, slice.content.size),
+                            // The slice content is used instead of getting the text directly from
+                            // the clipboard data because the pasted content could have already
+                            // been transformed by other ProseMirror plugins
+                            slice.content.textBetween(0, slice.content.size, '\n'),
```

Comment:
This was the other functional change, the main one actually, which makes sure that multi-line Markdown copied from VS Code work is split correctly (in Chromium browsers) by inserting aline feed character between nodes.